client/components/language-picker/index.jsx

```diff
@@ -174,13 +173,12 @@ export class LanguagePicker extends PureComponent {
 					</div>
 				</div>
 				{ this.renderModal( language.langSlug ) }
-				<QueryGeo />
 				<QueryLanguageNames />
 			</div>
 		);
 	}
 }
 
-export default connect( state => ( {
-	countryCode: getGeoCountryShort( state ),
-} ) )( localize( LanguagePicker ) );
+export default connect( () => ( { countryCode: requestGeoLocation().data } ) )(
```

Comment:
I'm thinking loudly here: a `mapStateToProps` function that doesn't use `state` and yet it works looks strange and suspicious. I didn't trust it until I understood the `HTTP_DATA_TICK` trick 😄 

There's an opportunity for a more friendly API that uses the popular "render prop" approach:
```js
<WithHttpData store="geo">
  { geo => <LanguagePicker countryCode="geo" /> }
</WithHttpData>
```

Reply:
on a similar note: if we made HttpData a Context Provider using React 16.3 APIs then we could avoid the data tick and still have the flexibility for either HoCs or render props.

Reply:
yeah I'm open to update later but I don't like the "render prop" approach because it seems to add a lot of code when we already have a place to put things like this that's normal. `connect()` is Redux-oriented but it's also serving a very specific purpose for us: what are the data objects that we need, what are the side-effecting functions that we need?

To me it makes sense to add into `mapStateToProps` because that's how I think many of us think about what role it's fulfilling; that or create a custom `connect()` wrapper but in my explorations there I've found it gets much more complicated that `getHttpData()` needs to be.